workflow-trace-viewer/src/jvmMain/kotlin/com/squareup/workflow1/traceviewer/App.kt

```diff
@@ -47,47 +63,78 @@ internal fun App(
   }
 
   Box(
-    modifier = modifier
+    modifier = modifier.onGloballyPositioned {
+      appWindowSize = it.size
+    }
   ) {
     fun resetStates() {
       selectedTraceFile = null
       selectedNode = null
       frameIndex = 0
-      workflowFrames.clear()
+      frameSize = 0
+      rawRenderPass = ""
+      active = false
+      nodeLocations.clear()
     }
 
     // Main content
     SandboxBackground(
+      appWindowSize = appWindowSize,
       sandboxState = sandboxState,
     ) {
       // if there is not a file selected and trace mode is live, then don't render anything.
-      val readyForFileTrace = traceMode is TraceMode.File && selectedTraceFile != null
-      val readyForLiveTrace = traceMode is TraceMode.Live
+      val readyForFileTrace = TraceMode.validateFileMode(traceMode)
+      val readyForLiveTrace = TraceMode.validateLiveMode(traceMode)
+
       if (readyForFileTrace || readyForLiveTrace) {
+        active = true
         RenderTrace(
           traceSource = traceMode,
           frameInd = frameIndex,
-          onFileParse = { workflowFrames.addAll(it) },
-          onNodeSelect = { node, prevNode ->
-            selectedNode = NodeUpdate(node, prevNode)
-          },
-          onNewFrame = { frameIndex += 1 }
+          onFileParse = { frameSize += it },
+          onNodeSelect = { selectedNode = it },
+          onNewFrame = { frameIndex += 1 },
+          onNewData = { rawRenderPass += "$it," },
+          storeNodeLocation = { node, loc -> nodeLocations[frameInd] += (node to loc) }
         )
       }
     }
 
-    FrameSelectTab(
-      frames = workflowFrames,
-      currentIndex = frameIndex,
-      onIndexChange = { frameIndex = it },
-      modifier = Modifier.align(Alignment.TopCenter)
-    )
-
-    RightInfoPanel(
-      selectedNode = selectedNode,
+    Column(
       modifier = Modifier
-        .align(Alignment.TopEnd)
-    )
+        .align(Alignment.TopCenter)
+        .padding(top = 8.dp),
+      verticalArrangement = Arrangement.spacedBy(8.dp),
+      horizontalAlignment = Alignment.CenterHorizontally
+    ) {
+      if (active) {
+        // Since we can jump from frame to frame, we fill in the map during each recomposition
+        if (nodeLocations.getOrNull(frameInd) == null) {
+          // frameSize has not been updated yet, so on the first frame, frameSize = nodeLocations.size = 0,
+          // and it will append a new map
+          while (nodeLocations.size <= frameSize) {
+            nodeLocations.add(mutableStateMapOf())
+          }
+        }
+
+        SearchBox(
+          nodes = nodeLocations[frameInd].keys.toList(),
+          onSearch = { name ->
+            sandboxState.scale = 1f
+            val node = nodeLocations[frameInd].keys.firstOrNull { it.name == name }
+            val newX = sandboxState.offset.x - nodeLocations[frameInd][node]!!.x + appWindowSize.width / 2
```

Comment:
Nit: `.getValue(key)` is a bit more idiomatic than `[key]!!`
```suggestion
            val newX = sandboxState.offset.x - nodeLocations[frameInd].getValue(node).x + appWindowSize.width / 2
```